### ISSUE TITLE
RenderManagerOpenGL::RenderEyeFinalize glBindFrameBuffer: OpenGL error GL_INVALID_OPERATION(1282)

### ISSUE THREAD

**Issue by Oneiros90** [sensics/OSVR-RenderManager#253]
Hi, I'm trying to run [this OpenGL example](https://github.com/sensics/OSVR-RenderManager/blob/master/examples/RenderManagerOpenGLExample.cpp). It works like one time out of five, resulting in a warning and an error most of the times ([screenshot](http://i.imgur.com/jIpvobe.png)).
Any idea why?

My setup:
- Windows 10
- Visual Studio 2015
- OSVR SDK 0.6.1339-g3139e8c
- Oculus Rift DK2 (Oculus SDK 0.7.0.0 + OSVR Oculus Plugin build78-0.7.0.0 x64)

I followed [these steps](http://wiki.osvr.com/display/DD/Developing+Client+Apps#DevelopingClientApps-AppDevelopmentProcess) to make the Visual Studio project and I just added `opengl32.lib`, `osvrClientKit.lib`, `osvrUtil.lib` and `osvrRenderManager.lib` to the linker.

I'm currently in Release mode (Debug doesn't work, it hangs on line 368...)
Thanks

**Comment by russell-taylor** [comment]
There has been some recent adjustment to where different rendering functions are happening in OpenGL.  If you don't have the current master checkout for RenderManager, some important fixes may be missing.  If you are using this, we'll need to try and reproduce here.

A reason for debug hanging could be due to the need to compile OSVR-Core and OSVR-RenderManager with the same Visual Studio settings (debug, etc).  This is because RenderManager currently uses the C++ API to talk with Core functions.

**Comment by Oneiros90** [comment]
I see. The problem is that it would be very difficult for me right now to c-make and generate the entire OSVR-Core and OSVR-RenderManager projects, fixing all the dependencies etc... using the sdk binaries was very handy...

Thank you anyway for your answer :)

**Comment by russell-taylor** [comment]
The most-recent compiled libraries at https://osvr.github.io/using/ should work together and also have the improvements I mentioned.  If the problem happens with them, we'll try to reproduce and fix.

**Comment by Oneiros90** [comment]
Yes, it happens with them. The error is [in this class at this line](https://github.com/sensics/OSVR-RenderManager/blob/master/osvr/RenderKit/RenderManagerOpenGL.cpp#L830).

**Comment by russell-taylor** [comment]
Unable to reproduce with the current master source of RenderManager repository and either a window-only server or an Oculus DK2 using the raw VRPN driver.  Will try with the released SDK to see if it happens there.

**Comment by russell-taylor** [comment]
Trying  0.6.1339-g3139e8c build 332, it fails to run about half the time.  When I finally caught it in a console window, it had that same error message.  Tried again with the master source version and it worked multiple times.  Looks like the SDK installer needs to be rebuilt with the current source.  Sorry this didn't happen automatically.  I'm asking for it to be triggered now.

**Comment by russell-taylor** [comment]
We've moved to a validated-build release approach, which is good, but in the process we stopped building automatic updates, which is less than optional.  I've now validated that the upcoming release (build 335) fixes this bug for me so hopefully will fix for you as well.  I'm also lobbying, it looks like successfully, to get a separate automatic build pushed out so that we don't have to wait for these in the future to test potential patches.
